linode/manager
# M3-5338: Notification Drawer refinements

## Description

This PR updates the UI for the notification drawer and it's empty state. The notification drawer is less of a drawer and is more consistent with the user menu. 

**Todo:**
- [x] Refactor `NotificationDrawer` to implement a Menu component (such as from the Reach package) instead of adding conditional logic inside of our `Drawer` component
- [x] Update namings from drawer to menu

**Tasks that will be addressed in a future PR:**
- [ ] Bold the username, action and entity name in the event messages inside the notification drawer

## How to test

Check the notification events view:
- Notifications but a few events (< 5 events)
- Notifications but a lot of events (turn on mocks)
- No notifications and no events (comment out code inside of `serverHandlers.ts`)

Test menu opening/closing:
- Opening the menu via another action other than the bell icon (such as clicking on a Linode's status when it is booting)
- Toggling the menu by double clicking the bell icon
- Clicking inside the menu should not close the menu
    - Unless you clicked a link inside the menu (such as `View all events`), then the menu should close

Other:
- Browsers such as Firefox and Safari
- Mobile view
